test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/DoubleSupplierTest.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.function
+
+import java.util.function.DoubleSupplier
+
+import org.junit.Assert._
+import org.junit.Test
+
+class DoubleSupplierTest {
+  import DoubleSupplierTest._
+  private val delta = 1e31
+
+  @Test def getAsDouble(): Unit = {
+    assertEquals(Double.NaN, makeSupplier(Double.NaN).getAsDouble(), delta)
+    assertEquals(Double.MinValue, makeSupplier(Double.MinValue).getAsDouble(), delta)
+    assertEquals(Double.MinPositiveValue, makeSupplier(Double.MinPositiveValue).getAsDouble(), delta)
+    assertEquals(Double.PositiveInfinity, makeSupplier(Double.PositiveInfinity).getAsDouble(), delta)
+    assertEquals(Double.NegativeInfinity, makeSupplier(Double.NegativeInfinity).getAsDouble(), delta)
+    assertEquals(1.234D, makeSupplier(1.234D).getAsDouble(), delta)
```

Comment:
I would keep only this one. There's no real point testing all sorts of values in this case, as it suggests that `DoubleSupplier` is somehow sensitive to particular values, which it most definitely is not.

---

Also, we always use a `0.0` delta. There are no computations, so there can be no loss of precision. Also we use lowercase `d` and `f` for double and float suffixes (but `L` for the Long suffix), as is typically widely acknowledged as best practice in Scala:
```suggestion
    assertEquals(1.234d, makeSupplier(1.234d).getAsDouble(), 0.0d)
```